controllers/osbuildenvconfig_controller.go

```diff
@@ -47,16 +130,419 @@ type OSBuildEnvConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildEnvConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	reqLogger := log.FromContext(ctx).WithValues("osbuildenvconfig", req.Name)
+	reqLogger.Info("Reconciling OSBuildEnvConfig")
 
-	// TODO(user): your logic here
+	instance, err := r.OSBuildEnvConfigRepository.Read(ctx, req.Name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found. Not a fatal error.
+			return ctrl.Result{}, nil
+		}
+		reqLogger.Error(err, "get failed for KubeFiler", "name", req.Name)
+		return ctrl.Result{Requeue: true}, err
+	}
 
-	return ctrl.Result{}, nil
+	// now that we have the resource. determine if its alive or pending deletion
+	if instance.GetDeletionTimestamp() != nil {
+		// its being deleted
+		if controllerutil.ContainsFinalizer(instance, osBuildOperatorFinalizer) {
+			// and our finalizer is present
+			return r.Finalize(ctx, reqLogger, instance)
+		}
+		return ctrl.Result{}, nil
+	}
+	// resource is alive
+	return r.Update(ctx, reqLogger, instance)
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *OSBuildEnvConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&osbuilderprojectflottaiov1alpha1.OSBuildEnvConfig{}).
+		For(&osbuildv1alpha1.OSBuildEnvConfig{}).
 		Complete(r)
 }
+
+func (r *OSBuildEnvConfigReconciler) Update(ctx context.Context, reqLogger logr.Logger, instance *osbuildv1alpha1.OSBuildEnvConfig) (ctrl.Result, error) {
+	reqLogger.Info("Updating state for OSBuildEnvConfig",
+		"name", instance.Name,
+		"UID", instance.UID,
+	)
+
+	changed, err := r.addFinalizer(ctx, instance)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	} else if changed {
+		reqLogger.Info("Added finalizer")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	_, changed, err = r.getOrCreateCertificate(
+		ctx,
+		reqLogger,
+		instance,
+		composerCertificateName,
+		[]string{
+			composerComposerAPIServiceName,
+			composerWorkerAPIServiceName,
+		},
+	)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	} else if changed {
+		reqLogger.Info("Created composer certificate")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	composerDeploymentParams := composerDeploymentParameters{
+		ComposerDeploymentNamespace:      conf.GlobalConf.WorkingNamespace,
+		ComposerDeploymentName:           composerDeploymentName,
+		ComposerImageName:                "quay.io/app-sre/composer",
```

Comment:
I think it can be useful to export those values to consts

Reply:
Done